timesketch/lib/analyzers/safebrowsing.py

```diff
@@ -0,0 +1,288 @@
+"""Sketch analyzer plugin for the Safe Browsing API."""
+from __future__ import unicode_literals
+
+import fnmatch
+import logging
+import re
+
+import pkg_resources
+import requests
+from flask import current_app
+
+from timesketch.lib.analyzers import interface
+from timesketch.lib.analyzers import manager
+
+
+class SafeBrowsingSketchPlugin(interface.BaseSketchAnalyzer):
+    """Sketch analyzer for Safe Browsing."""
+
+    NAME = 'safebrowsing'
+
+    # Safe Browsing API v4, threatMatches.find endpoint.
+    _SAFE_BROWSING_THREATMATCHING_ENDPOINT = (
+        'https://safebrowsing.googleapis.com/v4/threatMatches:find')
+
+    # Maximal number of URLs in a single Lookup request as per the
+    # Safe Browsing API documentation.
+    _SAFE_BROWSING_BULK_LIMIT = 500
+
+    # An optional file containing URL wildcards to be allow listed
+    # in a YAML file.
+    _URL_ALLOW_LIST_CONFIG = 'safebrowsing_allowlist.yaml'
+
+    # The keys to be added to the TS event from the ThreatMatch object
+    # we get from Safe Browsing API.
+    _SAFEBROWSING_ENTRY_KEEP = frozenset([
+        'platformType',
+        'threatType',
+    ])
+
+    # Used to find proper URLs in the 'url' entries of TS events.
+    _URL_BEGINNING_RE = re.compile(r'(http(s|):\/\/\S*)')
+
+    def __init__(self, index_name, sketch_id):
+        """Initialize The Sketch Analyzer.
+
+        Args:
+            index_name: Elasticsearch index name
+            sketch_id: Sketch ID
+        """
+        super(SafeBrowsingSketchPlugin, self).__init__(index_name, sketch_id)
+
+        self._safebrowsing_api_key = current_app.config.get(
+            'SAFEBROWSING_API_KEY')
+
+        self._google_client_id = current_app.config.get(
+            'SAFEBROWSING_CLIENT_ID',
+            'Timesketch',
+        )
+
+        self._google_client_version = current_app.config.get(
+            'SAFEBROWSING_CLIENT_VERSION',
+            pkg_resources.get_distribution('timesketch').version,
```

Comment:
you can now also just do

```
from timesketch import version

...

version.get_version()
```

to the get the TS version information, since it's moved outside of the package resource

Reply:
Wow, this is useful, so we won't need to import `pkg_resources` in this file. We'll follow up!